tests/baselines/reference/unusedWellKnownSymbolsInClass.errors.txt

```diff
@@ -0,0 +1,148 @@
+      }
+    
+      private static [Symbol.hasInstance](instance: unknown) {
+        return Array.isArray(instance);
+      }
+    
+      private get [Symbol.isConcatSpreadable]() {
+        return false
+      }
+      
+      private *[Symbol.iterator]() {
+        yield "hello";
+      }
+    
+      private get [Symbol.match]() {
+        return false;
+      }
+    
+      private *[Symbol.matchAll](str: string) {
+        for (const n of str.matchAll(/[0-9]+/g))
+          yield n[0];
+      }
+    
+      private [Symbol.replace](str: string) {
+        return `s/${str}/foo/g`;
+      }
+    
+      private [Symbol.search](str: string) {
+        return str.indexOf('foo');
+      }
+    
+      private [Symbol.split](str: string) {
+        return 'foo'
+      }
+    
+      private static get [Symbol.species]() { return Array; }
+    
+      private [Symbol.toPrimitive](hint: any) {
+        if (hint === 'number') {
+          return 42;
+        }
+        return null;
+      }
+    
+      private get [Symbol.toStringTag]() {
+        return 'foo';
+      }
+    
+      private get [Symbol.unscopables]() {
+        return {};
+      }
+    }
+    
+    export class WellknownSymbolProperties {
+      private [Symbol.asyncIterator] = function*() {
+        yield "hello";
+      }
+    
+      private static [Symbol.hasInstance] = (instance: unknown) => {
+        return Array.isArray(instance);
+      }
+    
+      private [Symbol.isConcatSpreadable] = false;
+      
+      private [Symbol.iterator] = function*() {
+        yield "hello";
+      }
+    
+      private [Symbol.match] = false;
+    
+      private [Symbol.matchAll] = function*(str: string) {
+        for (const n of str.matchAll(/[0-9]+/g))
+          yield n[0];
+      }
+    
+      private [Symbol.replace] = (str: string) => {
+        return `s/${str}/foo/g`;
+      }
+    
+      private [Symbol.search] = (str: string) => {
+        return str.indexOf('foo');
+      }
+    
+      private [Symbol.split] = (str: string) => {
+        return 'foo'
+      }
+    
+      private static [Symbol.species] = Array;
+    
+      private [Symbol.toPrimitive] = (hint: any) => {
+        if (hint === 'number') {
+          return 42;
+        }
+        return null;
+      }
+    
+      private [Symbol.toStringTag] = 'foo';
+    
+      private [Symbol.unscopables] = {};
+    }
+    
+    const MyIterator = Symbol();
+    export function testLocalSymbols() {
+      const Symbol = {
+        iterator: MyIterator
+      };
+    
+      return class Foo{
+        private *[Symbol.iterator]() {
+                 ~~~~~~~~~~~~~~~~~
+!!! error TS6133: '[Symbol.iterator]' is declared but its value is never read.
+          yield 1;
+        }
+      }
+    }
+    
+    const iteratorSymbol = Symbol.iterator;
+    export class C {
+      private *[iteratorSymbol]() {
+               ~~~~~~~~~~~~~~~~
+!!! error TS6133: '[iteratorSymbol]' is declared but its value is never read.

```

Comment:
We probably should not error here, but It is gonna take a lot of effort to fix.

Reply:
No, it is widened to an ordinary symbol.

https://ts-ast-viewer.com/#code/MYewdgzgLgBAllApgJwIZRMgygTwLYBGIANjALwy6EkB0CK6mA3AFCIAeADprMMahAgwAwjADeLGDE7I4AN3SIYAKgDa9NBmz4ixALoAKAJTiWkmAF8WFoA

![1649091681(1)](https://user-images.githubusercontent.com/6630042/161594705-c760e72e-388a-49b4-8fb2-a89dc1d38e3f.png)
![1649091715(1)](https://user-images.githubusercontent.com/6630042/161594781-1dcf54fa-eb6d-4b8c-a041-9914417c981b.png)

Reply:
Hm, definitely a bug - I've opened #53276 to track it, since all members of the global `SymbolConstructor` are, in fact, declared as `unique symbol`s nowadays.